### ISSUE TITLE
Implement `Default` Trait for `CreatePaymentIntent` Struct

### ISSUE THREAD

**Issue by matfndt** [arlyon/async-stripe#639]
### Is your feature request related to a problem? Please describe.

I am currently using this crate to implement a pre-authorization payment which requires some tweaking of the `createPaymentIntent`  struct. I am following the instructions in the stripe docs on ["Place a hold on a payment method"](https://docs.stripe.com/payments/place-a-hold-on-a-payment-method). The issue is more about improving ergonomics rather than adding new functionality. As of my knowledge for my use case I need to initialize the struct with all fields which is quite tedious.

### Describe the solution you'd like

I would implement the `Default` trait for `CreatePaymentIntent` to simplify the initialization process to allow only using subsets of the fields.

```rust
let params = CreatePaymentIntent {
        amount: 123, 
        currency: Currency::EUR,
        payment_method_types: Some(vec!["card".to_string()]),
        capture_method: Some(PaymentIntentCaptureMethod::Manual), // this is required for pre-auth
        metadata: Some(metadata),
        ..Default::default() // this doesn't work currently
    };
```
So some implementation such as this:
```
impl<'a> Default for CreatePaymentIntent<'a> {
    fn default() -> Self {
        CreatePaymentIntent {
            amount: 123, // some default value
            currency: Currency::USD, // some currency 
            // Add all other optional fields as None
        }
    }
}
```

**Comment by matfndt** [comment]
> Have you considered just using `CreatePaymentIntent::new` and then modifying it?
> 
> ```rust
> let mut params = CreatePaymentIntent::new(amount, currency);
> params.payment_method = ...
> params.customer = ...
> params.confirm = ...
> params.off_session = ...
> ```

Yes I did. I am not a big fan of having to mutate the instance. I know this works but I want to make it very clear that there is nothing changed to the payment intent in the creation. So it is more ergonomically but in my eyes not cleanly written. Thanks for the help though :)

**Comment by dyc3** [comment]
You could also try something like this to get the ergonomics you want:

```rust
let params  = {
  let mut params = CreatePaymentIntent::new(amount, currency);
  params.payment_method = ...
  params.customer = ...
  params.confirm = ...
  params.off_session = ...
  params
};
```

This way the mutability is limited only to the scope of your initialization.

**Comment by matfndt** [comment]
> You could also try something like this to get the ergonomics you want:
> 
> 
> 
> ```rust
> 
> let params  = {
> 
>   let mut params = CreatePaymentIntent::new(amount, currency);
> 
>   params.payment_method = ...
> 
>   params.customer = ...
> 
>   params.confirm = ...
> 
>   params.off_session = ...
> 
>   params
> 
> };
> 
> ```
> 
> 
> 
> This way the mutability is limited only to the scope of your initialization.

Thanks! I really like that idea of using scope and name aliasing. Never thought if that but really cool idea. Definitely going to use that in other code too.

**Comment by arlyon** [comment]
Closing due to #526 🎉 . We now have a nice builder API you can check out
